File: golden-paths/h2-enhancement/microservice/skeleton/go/health.go
package main

import (
	"net/http"
	"time"

	"github.com/gin-gonic/gin"
)

var startTime = time.Now()

func registerHealthRoutes(r *gin.Engine) {
	r.GET("/health", func(c *gin.Context) {
		c.JSON(http.StatusOK, gin.H{
			"status":  "healthy",
			"service": "${{ values.serviceName }}",
		})
	})

	r.GET("/ready", func(c *gin.Context) {
		c.JSON(http.StatusOK, gin.H{
			"status":         "ready",
			"uptime_seconds": int(time.Since(startTime).Seconds()),
		})
	})

	r.GET("/metrics", func(c *gin.Context) {
		c.JSON(http.StatusOK, gin.H{
			"service":        "${{ values.serviceName }}",
			"uptime_seconds": int(time.Since(startTime).Seconds()),
			"items_count":    len(items),
		})
	})
}

File: golden-paths/h2-enhancement/microservice/skeleton/nodejs/health.ts
import { Router } from "express";

const startTime = Date.now();

export const healthRouter = Router();

healthRouter.get("/health", (_req, res) => {
  res.json({ status: "healthy", service: "${{ values.serviceName }}" });
});

healthRouter.get("/ready", (_req, res) => {
  const uptimeSeconds = Math.round((Date.now() - startTime) / 1000);
  res.json({ status: "ready", uptime_seconds: uptimeSeconds });
});

healthRouter.get("/metrics", (_req, res) => {
  const uptimeSeconds = Math.round((Date.now() - startTime) / 1000);
  res.json({
    service: "${{ values.serviceName }}",
    uptime_seconds: uptimeSeconds,
  });
});
